ttnn/cpp/ttnn/operations/ccl/kernel_common/worker_sync_utils.hpp

```diff
@@ -151,3 +196,157 @@ FORCE_INLINE void advance_start_page_idx(
     }
 
 }
+
+
+struct MatmulOpReceiver {
+    uint32_t num_tensor_slices;
+    uint32_t num_directions = 2; // ASSUMPTION: Always 2 directions
+
+    bool wait_for_op_signal;
+    uint32_t num_transfers;
+    uint32_t ring_size;
+    uint32_t tensor_slice_shape_width; // In tiles
+    uint32_t output_page_offset;
+    uint32_t last_output_page_offset;
+
+    uint32_t num_blocks;
+    uint32_t num_blocks_per_slice;
+
+    // Used to track internal state
+    uint32_t ring_idxs[2];
```

Comment:
Could we use std::array here instead? Also use a named constant for size (Ie num directions from above)